packages/recommend-js/src/frequentlyBoughtTogether.tsx

```diff
@@ -71,15 +77,45 @@ function FrequentlyBoughtTogether<TObject>(
 export function frequentlyBoughtTogether<TObject>({
   container,
   environment = window,
+  itemComponent,
+  fallbackComponent,
+  headerComponent,
+  view,
+  children,
   ...props
-}: FrequentlyBoughtTogetherProps<TObject> & EnvironmentProps) {
-  const children = <FrequentlyBoughtTogether {...props} />;
+}: FrequentlyBoughtTogetherProps<TObject, Template> & EnvironmentProps) {
+  const vnode = (
+    <FrequentlyBoughtTogether<TObject, Template>
+      {...props}
+      view={view ? (viewProps) => view({ ...viewProps, html }) : undefined}
+      itemComponent={(itemComponentProps) =>
+        itemComponent({
+          ...itemComponentProps,
+          html,
+        })
+      }
+      headerComponent={
+        headerComponent
+          ? (headerProps) => headerComponent({ ...headerProps, html })
+          : undefined
+      }
+      fallbackComponent={
+        fallbackComponent
+          ? (fallbackProps) => fallbackComponent({ ...fallbackProps, html })
+          : undefined
+      }
+    >
+      {children
+        ? (childrenProps) => children({ ...childrenProps, html })
+        : undefined}
+    </FrequentlyBoughtTogether>
+  );
 
   if (!container) {
-    return children;
+    return vnode;
   }
 
-  render(children, getHTMLElement(container, environment));
+  render(vnode, getHTMLElement(container, environment));
 
-  return undefined;
+  return null;
```

Comment:
Why did we change this return type?

Reply:
It seems more accurate to return null when we have nothing in that case. Plus, we show examples where in the fallbackComponent we use one of our component in recommend but the types are incompatible as one is null and the other undefined.